### DISCUSSION TITLE
Financial donations and how they are used

### DISCUSSION THREAD

**Discussion by buhtz** [jmcgeheeiv/pyfakefs#839] [General]
Hello,

is it possible to financially donate to pyfakefs? Couldn't find something about it in the README.

Does pyfakefs use money in some way and how? It is nice to know for donating people how the money is used and how it can influence the quality and health of the project.

Kind

**Comment by mrbean-bremen** [comment]
No, there is is no sponsoring set up for `pyfakefs`, and at least I'm not planning for it. Currently, I'm the main contributor/maintainer to pyfakefs (after John McGehee, who has retired from the project for personal reasons), and I do that solely as a fun project in my free time. As I have a full-time job, and also contributing to other projects, the time I spend on pyfakefs does not amount to much, so I don't feel that I personally need any sponsoring. This may change in the future if other contributors get more active, though currently I don't see that. 

That being said, I currently do have a free subscription for the JetBrains tools, which they grant to main contributors for open source projects. I mainly use PyCharm for working on Python projects like pyfakefs (and also other tools like Rider and CLion for other projects), so I really appreciate that.

**Comment by buhtz** [comment]
Thanks for explaining. Background info like this are very useful for users and potential contributors. It is easier to keep in mind that maintainers are also human beings. You know what I mean? 😄 

What is about the "pytest-dev" connection? What is the value or benefit of that?

**Reply by mrbean-bremen** [reply]
Moving to an organization was an idea I had discussed some time ago with John McGehee, mainly for obvious reasons like a larger community, better visibility and the bus factor. `pytest-dev` seemed to fit best because `pyfakefs` is used more and more via the `pytest` fixture, and there is no other obvious matching organization. It also helped that I already had moved another repository there (I'm a fan of pytest). There has been some discussion in `pytest-dev` if `pyfakefs` really fits, as it is  bit of an outsider compared with the other dedicated pytest plugins, and it was accepted only on the second try - but I think it was a good move nevertheless. 

Recently, David Baird from Google has started being active in pyfakefs again (see #819), and he has stated that he is willing to steer the usage of pyfakefs at Google (which is substantial) in the direction of pytest. I also think that the usage via `pytest` will get more relevant in the future, and the visibility as a pytest plugin in `pytest-dev` facilitates that.

**Reply by buhtz** [reply]
Just a sidenote: I hope that pyfakefs will always be usable with vanilly unittest (without pytest), too. ;) I have my "problems" with the concept behind pytest. But I like its colorfull output when running unittest.

**Reply by mrbean-bremen** [reply]
Sure, there is no reason to remove unittest support. Technically, the support for both unittest and pytest is based on a wrapper around the `Patcher` implementation as created by John McGehee, so any changes will concern both test frameworks (and the `patchfs` decorator, which is another wrapper around `Patcher`).

**Comment by jmcgeheeiv** [comment]
Thank you so much for the offer, @buhtz. Just receiving the offer is gratifying; it makes us feel we are in the big leagues.

The software industry (particularly GitHub) supports pyfakefs with free stuff so we are not out-of-pocket. Accepting donations might be more trouble than not accepting donations. Of course should a giant wave of money show up, we might change our position.

Please consider contributing to projects that do solicit funds. I for example contribute to Wikimedia and LibreOffice. Tell them pyfakefs sent you.

**Reply by buhtz** [reply]
Thanks for explaining and givin an inside view to the project. I think it is a healthy project and I'm not worried anymore about it.
I highly depend on pyfakefs and that is why I think about its health. I try to avoid use of pyfakefs in my unittests but this comes with lot of disadvantages. A life without pyfakefs is possible but damn hard. :smile: 

I also tell my users donate to my depending projects because I also do not need money except for pizza and coffee. That is also why I asked about pyfakefs.
